### PR TITLE
[Google Drive] Add connector documentation

### PR DESCRIPTION
**Pull request by jedrazb**
## Closes https://github.com/elastic/enterprise-search-team/issues/5147

Add documentation to configure google drive connector.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes

### CODE REVIEW

**Review by wangch079 — COMMENTED** [review]
The Google Drive doc should go to [docs/reference](https://github.com/elastic/connectors-python/tree/main/docs/reference), before it's moved to the official guide.

You can follow the pattern of [the Dropbox connector](https://github.com/elastic/connectors-python/blob/main/docs/reference/dropbox.md)

File: docs/reference/google_drive.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+## Usage
+
+To use this connector as a **connector client**, use the **Google Drive** tile from the connectors list or **Customized connector** workflow.
+
+For additional operations, see [Usage](https://www.elastic.co/guide/en/enterprise-search/master/connectors-usage.html).
+
+## Configuring the Google Drive Connector:
+
+You need to configure the Google Drive connector before syncing the data. Fot this you need to create a [service account](https://cloud.google.com/iam/docs/service-account-overview) with appropriate access to Google Drive API.
```
Comment (navarone-feekery) [inline docs/reference/google_drive.md:19]:
```suggestion
You need to configure the Google Drive connector before syncing any data. For this you need to create a [service account](https://cloud.google.com/iam/docs/service-account-overview) with appropriate access to Google Drive API.
```

File: docs/reference/google_drive.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+To get started, log into [Google Cloud Platform](cloud.google.com) and go to the `Console`.
+
+1. **Create a Google Cloud Project.** Give your project a name, change the project ID and click the Create button.
+
+2. **Enable Google APIs.** Choose APIs & Services from the left menu and click on `Enable APIs and Services`. You need to enable the Drive API.
+
+3. **Create a Service Account.** In the `APIs & Services` section, click on `Credentials` and click on `Create credentials` to create a service account. Give your service account a name and a service account ID. This is like an email address and will be used to identify your service account in the future. Click `Done` to finish creating the service account.
+
+4. **Create a Key File**. In the Cloud Console, go to IAM and Admin > Service accounts page. Click the email address of the service account that you want to create a key for. Click the `Keys` tab. Click the `Add key` drop-down menu, then select `Create new key`. Select JSON as the Key type and then click `Create`. This will download a JSON file that will contain the service account credentials.
```
Comment (navarone-feekery) [inline docs/reference/google_drive.md:29]: Nit: This feels like it could be a dot-point list of actions. It's hard to follow progress as a paragraph.

File: docs/reference/google_drive.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+The service account JSON file includes the following information:
+
+- `Client ID`: A unique identifier for the service account.
+- `Client Email`: The email address associated with the service account.
+- `Private Key`: A private key used to authenticate requests made by the service account.
+- `Private Key ID`: An identifier for the private key.
+- `Token URI`: The URI where the service account can obtain an access token.
+- `Project ID`: The ID of the Google Cloud project associated with the service account.
+- `Auth URI`, Token URI, Auth Provider X509 Cert URL: URLs used for authentication and authorization purposes.
```
Comment (navarone-feekery) [inline docs/reference/google_drive.md:47]:
```suggestion
- `Auth URI`, `Token URI`, `Auth Provider X509 Cert URL`: URLs used for authentication and authorization purposes.
```

**Review by navarone-feekery — COMMENTED** [review]
Looks good! Left some comments :)

File: docs/reference/google_drive.md (hunk 2 above)
Comment (jedrazb) [reply docs/reference/google_drive.md]: Sure, will split this up.

**Review by navarone-feekery — APPROVED** [review]
Nice one!
